concordium-consensus/src/Concordium/TimeMonad.hs

```diff
@@ -22,6 +22,17 @@ class Monad m => TimeMonad m where
     default currentTime :: MonadIO m => m UTCTime
     currentTime = liftIO getCurrentTime
 
+    -- |Measure the time of a given action m a.
+    -- Note. One has to remember to factor in laziness when
+    -- reasoning about the returned 'NominalDiffTime'.
+    measureTime :: m a -> m (NominalDiffTime, a)
```

Comment:
WHy is this a typeclass method as opposed to just a function that works for any TimeMonad?

It is not overridden anywhere so I don't think it should be a method.